Prose/ProseLib/Sources/JoinChatFeature/AddMemberSheet.swift

```diff
@@ -0,0 +1,256 @@
+//
+// This file is part of prose-app-macos.
+// Copyright (c) 2022 Prose Foundation
+//
+
+import AppLocalization
+import Assets
+import ComposableArchitecture
+import ProseCoreTCA
+import ProseUI
+import SwiftUI
+
+private let l10n = L10n.AddMember.self
+
+// MARK: - View
+
+public struct AddMemberSheet: View {
+  public typealias ViewState = AddMemberSheetState
+  public typealias ViewAction = AddMemberSheetAction
+
+  let store: Store<ViewState, ViewAction>
+
+  public init(store: Store<ViewState, ViewAction>) {
+    self.store = store
+  }
+
+  public var body: some View {
+    WithViewStore(self.store) { viewStore in
+      VStack(spacing: 16) {
+        VStack(alignment: .leading) {
+          Text(verbatim: l10n.Header.title)
+            .font(.headline)
+          Text(verbatim: l10n.Header.subtitle)
+            .font(.subheadline)
+        }
+        .fixedSize(horizontal: false, vertical: true)
+        .frame(maxWidth: .infinity, alignment: .leading)
+        // TODO: [Rémi Bardon] Refactor `BasicAuthView.chatAddressHelpButton` so we can add the button here too.
+        // TODO: [Rémi Bardon] Maybe we could even create a `JIDField`, which supports all of this out of the box.
+        TextField(
+          l10n.TextField.label,
+          text: viewStore.binding(\.$text),
+          prompt: Text(verbatim: l10n.TextField.prompt)
+        )
+        .textFieldStyle(.roundedBorder)
+        .onSubmit(of: .text) { viewStore.send(.submitTapped) }
+        .frame(maxWidth: .infinity, alignment: .leading)
+        Self.infoMessage(for: viewStore.info)
+          .symbolVariant(.fill)
+          .fixedSize(horizontal: false, vertical: true)
+          .redacted(reason: viewStore.isProcessing ? .placeholder : [])
+        HStack {
+          Button { viewStore.send(.cancelTapped) } label: {
+            Text(verbatim: l10n.CancelAction.label)
+          }
+          .buttonStyle(.bordered)
+          Button { viewStore.send(.submitTapped) } label: {
+            if case .unknown = viewStore.info {
+              Label(l10n.InviteAction.label, systemImage: "link.circle.fill")
+            } else {
+              Label(l10n.SendRequestAction.label, systemImage: "person.crop.circle.fill.badge.plus")
+            }
+          }
+          .buttonStyle(.borderedProminent)
+          .disabled(!viewStore.isFormValid)
```

Comment:
I'd expect the submit button also to be disabled when `isProcessing` is true.

Reply:
Oh yes, that was a mistake in my preview, this is how I intended it too.